### PR TITLE
Add some mlab alternatives

### PR DESCRIPTION
**Pull request by dstansby**
Note I've changed `np` to `numpy` to make things clearer. Partially solves #10382.

### CODE REVIEW

File: lib/matplotlib/mlab.py [hunk 1]
```diff
@@ -2287,7 +2287,7 @@ def log2(x, ln2=math.log(2.0)):
             return len(bin_n)
 
 
-@cbook.deprecated("2.2")
+@cbook.deprecated("2.2", 'numpy.mod(n, 2)')
```
Comment (phobson) [inline lib/matplotlib/mlab.py]: This isn't quite right. 

From: 
```python
import numpy
from matplotlib import mlab
N = [37, 48, 64]
print('numpy:\t', list(map(lambda x: numpy.mod(x, 2), N)))
print('mlab:\t', list(map(mlab.ispower2, N)))
```

I get:
```
numpy:	 [1, 0, 0]
mlab:	 [0, 0, 6]
```

Comment (dstansby) [reply lib/matplotlib/mlab.py]: Oh yeah, that function doesn't quite do what it says on the tin (not that my alternative actually worked for determining if a number is a power of 2!)

**Comment by jklymak** [comment]
Are none of these used internally? If they are, they should be removed while deprecating?

**Comment by QuLogic** [comment]
Should already have been removed in the previous PR.

**Comment by dstansby** [comment]
Yeah, my previous big PR removed all the stuff that's deprecated from internal code.

**Pull request merged**